Use special "displayName" for zones and show full name in tooltip

madsnedergaard opened the pull request:
## Issue

Part 1 of #5837

## Description

This makes it possible show to a shorter name for a zone, while having the full name shown in a tooltip. This means we can use "ERCOT" for Texas in tooltips, while still having the actual name available to anyone curious :)

I will follow up with a second PR with the actual translation updates.

### Preview



### Double check

- [ ] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

VIKTORVAV99 commented:
@madsnedergaard I really like this idea but can we try and get some of the other open PRs that affects translations first? In theory this PR shouldn't cause merge conflicts if we just add names but you never know; and I don't want merge conflicts in all the PRs, including this one 😅

https://github.com/electricitymaps/electricitymaps-contrib/pull/6227 should be ready to go, and then once that is merged I should be able to finish up the translations in https://github.com/electricitymaps/electricitymaps-contrib/pull/6443 without merge conflicts (and doing the same work twice).

VIKTORVAV99 commented on web/src/translation/translation.ts:
```diff
@@ -32,29 +32,18 @@ export const translateIfExists = (key: string) => {
-export const getZoneName = (zoneCode: string) =>
-  translateIfExists(`zoneShortName.${zoneCode}.zoneName`);
+export const getFullZoneName = (zoneCode: string) => {
+  return translateIfExists(`zoneShortName.${zoneCode}.zoneName`);
+};
+export const getZoneName = (zoneCode: string) => {
+  const displayName = translateIfExists(`zoneShortName.${zoneCode}.displayName`);
+  const fullName = getFullZoneName(zoneCode);
+  return displayName || fullName;
+};
```
If you want to do a micro optimisation size wise you can extract the `zoneShortName.${zoneCode}.zoneName` to arrow function to avoid the extra string size from declaring it twice 😅

madsnedergaard commented:
Ready for review @VIKTORVAV99, I have also added a single zone for now so we can better test the functionality :)

VIKTORVAV99 commented on web/src/translation/translation.ts:
```diff
@@ -4,29 +4,18 @@ export const translateIfExists = (key: string) => {
+  return translateIfExists(`zoneShortName.${zoneCode}.zoneName`);
+};
+export const getZoneName = (zoneCode: string) => {
+  const displayName = translateIfExists(`zoneShortName.${zoneCode}.displayName`);
```
Have we tested that this translateIfExists works properly with the i18n system?

I can't remember if I did that or not when migrating.

madsnedergaard replied:
We are already using it, so I sure hope that you did test it 😅 
But should be fine, otherwise current production app would be broken

VIKTORVAV99 replied:
So do I! 😅

But remember who did the thorough review 😉

VIKTORVAV99 commented on web/src/features/panels/zone/ZoneHeaderTitle.tsx:
```diff
@@ -14,12 +14,18 @@ interface ZoneHeaderTitleProps {
+
 export default function ZoneHeaderTitle({ zoneId }: ZoneHeaderTitleProps) {
-  const title = getZoneName(zoneId);
+  const zoneNameShort = getZoneName(zoneId);
```
There is no guarantee this is actually shorter since it only would be if there is a display name property for the zone.

I think we should just rename it to zoneDisplayName (and the other to zoneFullName).

VIKTORVAV99 reviewed (commented):
The styling here don't match the details page, is this something we want to adress in here?
<img width="377" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/30777521/41313bcb-debd-488f-8aa6-73cc6cb45c1c">
It's more noticeable now since the title doesn't get truncated.

Perhaps a little out of scope but could you increase the title length a tiny bit here as well if you change the tooltip?
<img width="377" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/30777521/b2ee9da0-ac81-4ccb-981c-b5be8a3e28de">

madsnedergaard commented on web/src/translation/translation.ts:
```diff
@@ -4,29 +4,18 @@ export const translateIfExists = (key: string) => {
+};
+export const getZoneName = (zoneCode: string) => {
+  const displayName = translateIfExists(`zoneShortName.${zoneCode}.displayName`);
+  const fullName = getFullZoneName(zoneCode);
+  return displayName || fullName;
```
Yeeeah, I thought about that too... Done now!

madsnedergaard commented:
> The styling here don't match the details page, is this something we want to adress in here?

Did some changes now.

> Perhaps a little out of scope but could you increase the title length a tiny bit here as well if you change the tooltip?

Fixed (and added Storybook stories as it was a pain to develop on) :)

VIKTORVAV99 reviewed (approved):
One tiny thing, otherwise this looks good to me!

madsnedergaard commented on web/src/features/map/MapTooltip.stories.tsx:
```diff
@@ -0,0 +1,86 @@
+}
+
+const meta: Meta<typeof TestWrapper> = {
+  title: 'Modal/MapTooltip',
```
Good catch! :D
